### ISSUE TITLE
Error: required Datastore.Spec entry missing form config file

### ISSUE THREAD

**Issue by maxkerp** [ipfs/js-ipfs-repo#163]
Hi there :)

### TL;DR
`js-ipfs-repo` seems to miss some entries in the `config` file.

### What
I'm working with `orbit-db-cli`  and trying to examine the `ipfs repo` with `go-ipfs` results in this error message. `js-ipfs` doesn't care about it and works fine.

I'd liked to be able to examine the `repo` with `go-ipfs` (since it has a bit more features ), but adding the default `config` file to it, does not solve the issue.

Any ideas what I can do? Adding the entry to the config file should fix the issue, but I might made some stupid mistake I'm not aware of.

For more background see [this issue](https://github.com/orbitdb/orbit-db-cli/issues/16).

**Comment by diasdavid** [comment]
@maxkerp is this still an issue with the latest ipfs-repo?

**Comment by maxkerp** [comment]
I created a new temporary jsipfs repo under `/tmp` and it still seems to be the case.

```bash
/tmp $ IPFS_PATH=./ipfs/ jsipfs add data.json
added QmYvuL72a7qZJq6EzXAzGix1FsJ1ZUjgAXvAuq2795bznw data.json

/tmp $ IPFS_PATH=./ipfs/ jsipfs cat QmYvuL72a7qZJq6EzXAzGix1FsJ1ZUjgAXvAuq2795bznw
"{\"hash\":null,\"id\":\"QmXGrQZ8vPqwsCLHL9DhQEYLQb13n96oUtLatteh1zNk89\",\"payload\":{\"op\":\"PUT\",\"key\":\"002\",\"value\":{\"key\":\"002\",\"name\":\"Daniel\"}},\"next\":[\"QmRvhdhy5GQzs7Rd9KRh1x8GSCEXooFBa9GqbZvixnn9s1\"],\"v\":0,\"clock\":{\"id\":\"QmXGrQZ8vPqwsCLHL9DhQEYLQb13n96oUtLatteh1zNk89\",\"time\":2},\"sig\":\"304502203c5e6e26377e46b49f5bccdf6bd066f31314c0caa90fd05550f1616715d4bb38022100800665bc58d7639a5403dcab6dc52428a281219f46b66523e01f74a9e956c465\",\"key\":\"04fbfa074a34326a6099d1c1a05109b610687e5066c2ba4944105e15fdc74261ac391520d47fd2064eb6f6458224f2a66a89f03215bc3d51c1922a15db5606e6cc\"}"

/tmp $ IPFS_PATH=./ipfs/ ipfs cat QmYvuL72a7qZJq6EzXAzGix1FsJ1ZUjgAXvAuq2795bznw
Error: required Datastore.Spec entry missing form config file

/tmp $ jsipfs --version
0.29.2

/tmp $ ipfs --version
ipfs version 0.4.14
```
I created the repo with `/tmp $ IPFS_PATH=./ipfs/ jsipfs init`

**Comment by vasco-santos** [comment]
Basically `go-ipfs` requires the existence of  `datastore_spec` file [here](https://github.com/ipfs/go-ipfs/blob/07feeec9c4f40f59efb57d7e16739a35b4c42d5b/repo/fsrepo/fsrepo.go#L403).

`js-ipfs-repo` does not create this file and that's why there are problems with the repo interoperability in both [interop#26](https://github.com/ipfs/interop/pull/26) and [interop#28](https://github.com/ipfs/interop/pull/28).
